utils/clone_repo.py

```diff
@@ -47,4 +48,27 @@ def clone_joern_and_checkout(joern_branch, boost=False):
         repo.git.checkout(joern_branch)
         print("Checked out branch")
     except Exception as e:
-        print(f"Error while cloning joern: {e}")
\ No newline at end of file
+        print(f"Error while cloning joern: {e}")
```

Comment:
Can we fail the action here?

Reply:
Yes we can

Reply:
@ankit-privado Are we catching the error while doing the checkout for the same branch as that of joern fork? @karan-batavia if that's the case then we need to proceed further and use the `dev` branch if the respective branch doesn't exist.

Reply:
Cool, if that's the expected behaviour let's add a log there and use the `dev` branch instead of failing the action.

Reply:
Just realised, there is no `dev` branch for `joern`, @pandurangpatil

Reply:
No.

Reply:
I had discussions with @karan-batavia, checkout works only when branch is present in Privado-core otherwise it will pick the default Dev branch for head. 

Cc @pandurangpatil